packages/compass-collection/src/components/collection/collection.tsx

```diff
@@ -59,37 +73,41 @@ type CollectionProps = {
     id: string;
   };
   scopedModals: any[];
+  stats: StatsObject;
 };
 
-const Collection: React.FunctionComponent<CollectionProps> = ({
-  namespace,
-  isReadonly,
-  isTimeSeries,
-  statsPlugin,
-  statsStore,
-  editViewName,
-  sourceReadonly,
-  sourceViewOn,
-  selectOrCreateTab,
-  pipeline,
-  sourceName,
-  activeSubTab,
-  id,
-  queryHistoryIndexes,
-  tabs,
-  views,
-  localAppRegistry,
-  globalAppRegistry,
-  changeActiveSubTab,
-  scopedModals,
-}: CollectionProps) => {
+const Collection: React.FunctionComponent<CollectionProps> = (
+  props: CollectionProps
+) => {
+  const {
+    darkMode,
+    namespace,
+    isReadonly,
+    isTimeSeries,
+    editViewName,
+    sourceReadonly,
+    sourceViewOn,
+    selectOrCreateTab,
+    pipeline,
+    sourceName,
+    activeSubTab,
+    id,
+    queryHistoryIndexes,
+    tabs,
+    views,
+    localAppRegistry,
+    globalAppRegistry,
+    changeActiveSubTab,
+    scopedModals,
+    stats,
+  } = props;
   useEffect(() => {
     if (tabs && tabs.length > 0) {
       track('Screen', {
         name: trackingIdForTabName(tabs[activeSubTab] || 'Unknown'),
       });
     }
-  }, []);
+  });
```

Comment:
Giving `useEffect` no second argument acts as both componentDidMount and componentDidUpdate, as in it runs first on mount and then on every re-render. Do we want to keep this using the `[]`? I'm not sure I follow the change. If we want to track the sub tab changes maybe we make it use `activeSubTab` and `tabs` as dependencies?

Reply:
Lint complaints with []: `React Hook useEffect has missing dependencies: 'activeSubTab' and 'tabs'. Either include them or remove the dependency array.`

What would be the proper approach?

Reply:
Lint is showing that error because function is using the `activeSubTab` and `tabs` props inside of it. Because of the function's scope those variables could be stale in the useEffect, or not the current, correct value if we pass only `, []` because the function is only set on first mount. This might be alright if we only want it to run once, but in this case do we want the tracking to occur every time the `activeSubTab` changes? 
If we want the tracking event every time the `activeSubTab` or `tabs` change we can add them to the dependency list like this:
```suggestion
  }, ['activeSubTab', 'tabs']);
```

More information on how `useEffect` and other hooks use these dependencies and why the `exhaustive-deps` lint shows up can be found here: https://reactjs.org/docs/hooks-reference.html#conditionally-firing-an-effect

Reply:
If we want the event to only occur when the `activeSubTab` changes and not the `tabs` we could use another hook, `useMemo` to track the active subtab name and then add that new variable to this dependency list.
Something like:

```ts
const activeSubtabName = useMemo(() => trackingIdForTabName((tabs && tabs[activeSubTab]) || 'Unknown'), ['activeSubTab', 'tabs']);
```

```suggestion
  }, [activeSubtabName]);
```

Reply:
Returned []!